sklearn/model_selection/tests/test_validation.py

```diff
@@ -921,7 +921,7 @@ def check_cross_val_predict_with_method(est):
     X, y = shuffle(X, y, random_state=0)
     classes = len(set(y))
 
-    kfold = KFold(len(iris.target))
```

Comment:
@MechCoder Thanks, I think it may be OK to use KFold here. 
(1)The aim of the function here is to test parameter `method` in `cross_val_predict`, so we only need to choose a cross validation method.
(2)The iris dataset is quite balanced(each class 50 instances).
(3)Other functions in the file also use iris dataset and KFold for cross validation.